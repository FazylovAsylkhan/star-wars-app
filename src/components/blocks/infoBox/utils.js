function getArrPlanetsData(objectData) {
  const arrInfoHomeworld = [
    `Population: ${objectData?.population}`,
    `Diameter: ${objectData?.diameter}`,
    `Rotation period: ${objectData?.rotation_period}`,
    `Climate: ${objectData?.climate}`,
    `Gravity: ${objectData?.gravity}`,
    `Terrain: ${objectData?.terrain}`,
    `Surface water: ${objectData?.surface_water}`,
    `Orbital period: ${objectData?.orbital_period}`,
  ];
  return arrInfoHomeworld;
}
function getArrStarshipsData(objectData) {
  const arrInfoHomeworld = [
    `model: ${objectData?.model}`,
    `MGLT: ${objectData?.MGLT}`,
    `Consumables: ${objectData?.consumables}`,
    `Crew: ${objectData?.crew}`,
    `Hyperdrive rating: ${objectData?.hyperdrive_rating}`,
    `Length: ${objectData?.hyperdrive_rating}`,
    `Manufacturer: ${objectData?.manufacturer}`,
    `Max atmosphering speed: ${objectData?.max_atmosphering_speed}`,
    `Passengers: ${objectData?.passengers}`,
    `Starship class: ${objectData?.starship_class}`,
  ];
  return arrInfoHomeworld;
}
function getArrPeopleData(objectData) {
  const arrInfoPerson = [
    `Gender: ${objectData?.gender}`,
    `Height: ${objectData?.height}`,
    `Mass: ${objectData?.mass}`,
    `Hair color: ${objectData?.hair_color}`,
    `Skin color: ${objectData?.skin_color}`,
    `Eye color: ${objectData?.eye_color}`,
    `Birth year: ${objectData?.birth_year}`,
  ];
  return arrInfoPerson;
}

export function getArrHomeworldData(objectData) {
  const arrInfoHomeworld = [
    `Homeworld: ${objectData?.name}`,
    `Population: ${objectData?.population}`,
    `Diameter: ${objectData?.diameter}`,
    `Rotation period: ${objectData?.rotation_period}`,
  ];
  return arrInfoHomeworld;
}

export function getRelativeUrl(url) {
  return url.split("/").splice(4);
}
export function getHandlerDataFor(namePage) {
  switch (namePage) {
    case "people":
      return getArrPeopleData;
    case "planets":
      return getArrPlanetsData;
    case "starships":
      return getArrStarshipsData;
    default:
      return getArrHomeworldData;
  }
}
