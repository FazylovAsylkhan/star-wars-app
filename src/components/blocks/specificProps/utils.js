function getAllExistSpecificProps(data) {
  return [
    { name: "Starships", values: data.starships },
    { name: "Vehicles", values: data.vehicles },
    { name: "Films", values: data.films },
    { name: "Species", values: data.species },
    { name: "Residents", values: data.residents },
    { name: "Pilots", values: data.pilots },
  ];
}

function filterProps(props) {
  const filteredProps = props.filter((prop) => {
    const { values } = prop;
    const IsExitValues = values !== undefined;
    const IsNotEmptyValues = values?.length > 0;

    return IsExitValues && IsNotEmptyValues ? prop : null;
  });

  return filteredProps;
}

const getFilteredProps = (data) => {
  let result = null;
  if (data) {
    const allExistSpecificProps = getAllExistSpecificProps(data);
    result = filterProps(allExistSpecificProps);
  }
  return result;
};

export default getFilteredProps;
