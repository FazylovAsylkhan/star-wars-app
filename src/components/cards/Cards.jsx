import React from "react";
import Card from "./card";
import s from "./cards.module.scss";

function Cards({ data }) {
  const { results } = data;

  return (
    <div data-testid="cards" className={s.cards}>
      {results.map((result, index) => (
        <Card key={result.name} data={result} idPersonalPage={`${index + 1}`} />
      ))}
    </div>
  );
}

export default Cards;
