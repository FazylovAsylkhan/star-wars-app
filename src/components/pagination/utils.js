import s from "../button/button.module.scss";

function addActiveClass(url = null) {
  const currentNumberOfPage = url ? url.split("=")[1] : "1";
  const pagination = document.getElementById("pagination");
  const btns = Array.from(pagination.getElementsByTagName("button"));
  btns.forEach((btn) => {
    btn.classList.remove(s.activeBtn);
    if (btn.textContent === currentNumberOfPage) btn.classList.add(s.activeBtn);
  });
}

export default addActiveClass;
