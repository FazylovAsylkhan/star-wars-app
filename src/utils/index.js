export const transformSubDataToArray = (obj) =>
  obj.results.map((item) => Object.entries(item));

export const transformDataToArray = (item) => Object.entries(item);

export const getIdPageFrom = (list) => {
  let id = 0;
  if (list.length) {
    const url = list[list.length - 1][1];
    const arr = url.split("/");
    id = arr[arr.length - 2];
  }

  return id;
};
