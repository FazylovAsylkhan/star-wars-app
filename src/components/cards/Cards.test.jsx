/* eslint-disable no-undef */
import React from "react";
import { screen } from "@testing-library/react";
import { renderWithProvidersAndRouter } from "../../tests/testUtils";
import { transformSubDataToArray } from "../../utils";
import testData from "../../tests/testData";
import Cards from "./Cards";

describe("Cards component works", () => {
  const handledData = transformSubDataToArray(testData);

  it("should render Cards component", async () => {
    renderWithProvidersAndRouter(
      <Cards results={handledData} data={testData} />
    );
    const cards = screen.getByTestId("cards");
    expect(cards).toBeInTheDocument();
    expect(cards).toMatchSnapshot();
  });

  it("number of cards should be 10", () => {
    renderWithProvidersAndRouter(
      <Cards results={handledData} data={testData} />
    );
    const numberOfCards = screen.getAllByRole("list").length;
    expect(numberOfCards).toBe(3);
  });
});
