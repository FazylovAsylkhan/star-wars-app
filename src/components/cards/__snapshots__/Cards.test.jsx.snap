// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component works should render Cards component 1`] = `
<div
  class="wrapper"
  data-testid="cards"
>
  <div
    class="pagination"
    data-testid="pagination"
  >
    <button
      class="button dark"
      type="button"
    >
      Previous
    </button>
    <button
      class="button light"
      type="button"
    >
      1
    </button>
    <button
      class="button light"
      type="button"
    >
      2
    </button>
    <button
      class="button light"
      type="button"
    >
      3
    </button>
    <button
      class="button light"
      type="button"
    >
      4
    </button>
    <button
      class="button light"
      type="button"
    >
      5
    </button>
    <button
      class="button light"
      type="button"
    >
      6
    </button>
    <button
      class="button light"
      type="button"
    >
      7
    </button>
    <button
      class="button light"
      type="button"
    >
      8
    </button>
    <button
      class="button light"
      type="button"
    >
      9
    </button>
    <button
      class="button light"
      type="button"
    >
      Next
    </button>
  </div>
  <div
    class="cards"
  >
    <div
      class="card"
      data-testid="card"
    >
      <ul
        class="list"
      >
        <li>
          <b>
            name
            : 
          </b>
          Luke Skywalker 1
        </li>
        <li>
          <b>
            birth_year
            : 
          </b>
          19BBY
        </li>
        <li>
          <b>
            created
            : 
          </b>
          2014-12-09T13:50:51.644000Z
        </li>
        <li>
          <b>
            edited
            : 
          </b>
          2014-12-20T21:17:56.891000Z
        </li>
        <li>
          <b>
            eye_color
            : 
          </b>
          blue
        </li>
        <li>
          <b>
            films
            : 
          </b>
          https://swapi.dev/api/films/1/
          https://swapi.dev/api/films/2/
          https://swapi.dev/api/films/3/
          https://swapi.dev/api/films/6/
        </li>
        <li>
          <b>
            gender
            : 
          </b>
          female
        </li>
        <li>
          <b>
            hair_color
            : 
          </b>
          blond
        </li>
      </ul>
      <a
        class="button primary"
        href="/1"
      >
        More ...
      </a>
    </div>
    <div
      class="card"
      data-testid="card"
    >
      <ul
        class="list"
      >
        <li>
          <b>
            name
            : 
          </b>
          Luke Skywalker2
        </li>
        <li>
          <b>
            birth_year
            : 
          </b>
          19BBY
        </li>
        <li>
          <b>
            created
            : 
          </b>
          2014-12-09T13:50:51.644000Z
        </li>
        <li>
          <b>
            edited
            : 
          </b>
          2014-12-20T21:17:56.891000Z
        </li>
        <li>
          <b>
            eye_color
            : 
          </b>
          blue
        </li>
        <li>
          <b>
            films
            : 
          </b>
          https://swapi.dev/api/films/1/
          https://swapi.dev/api/films/2/
          https://swapi.dev/api/films/3/
          https://swapi.dev/api/films/6/
        </li>
        <li>
          <b>
            gender
            : 
          </b>
          male
        </li>
        <li>
          <b>
            hair_color
            : 
          </b>
          blond
        </li>
      </ul>
      <a
        class="button primary"
        href="/2"
      >
        More ...
      </a>
    </div>
    <div
      class="card"
      data-testid="card"
    >
      <ul
        class="list"
      >
        <li>
          <b>
            name
            : 
          </b>
          Luke Skywalker 3
        </li>
        <li>
          <b>
            birth_year
            : 
          </b>
          19BBY
        </li>
        <li>
          <b>
            created
            : 
          </b>
          2014-12-09T13:50:51.644000Z
        </li>
        <li>
          <b>
            edited
            : 
          </b>
          2014-12-20T21:17:56.891000Z
        </li>
        <li>
          <b>
            eye_color
            : 
          </b>
          blue
        </li>
        <li>
          <b>
            films
            : 
          </b>
          https://swapi.dev/api/films/1/
          https://swapi.dev/api/films/2/
          https://swapi.dev/api/films/3/
          https://swapi.dev/api/films/6/
        </li>
        <li>
          <b>
            gender
            : 
          </b>
          male
        </li>
        <li>
          <b>
            hair_color
            : 
          </b>
          blond
        </li>
      </ul>
      <a
        class="button primary"
        href="/3"
      >
        More ...
      </a>
    </div>
  </div>
</div>
`;
